components/scream/src/share/atm_process/atmosphere_diagnostic.hpp

```diff
@@ -43,14 +43,16 @@ class AtmosphereDiagnostic : public AtmosphereProcess
   // Getting the diagnostic output
   Field get_diagnostic () const;
 
-  // Avoid shadowing the base class method, but allow calling without passing a dummy dt
-  using AtmosphereProcess::run;
-  void run () { this->run_impl(0); } // Pass null dt, but should not be used anyways.
-
   void set_computed_field (const Field& f) final;
   void set_computed_group (const FieldGroup& group) final;
+
+  void compute_diagnostic ();
 protected:
 
+  virtual void compute_diagnostic_impl () = 0;
+
+  void run_impl (const int /* dt */);
```

Comment:
is it safe to say that we don't actually ever use `run_impl` now?

Reply:
Well, probably. But until we refactor the diag class, making it independent of AtmosphereProcess, we don't want to risk someone calling the run methods. Since it's easy to fwd a call to run to a call to compute_diagnostic, we might as well do it.

Long term, I think Diagnostic should not be an atm process.

Reply:
I agree